src/python/pants/backend/jvm/tasks/classpath_products.py

```diff
@@ -329,7 +334,15 @@ def _add_excludes_for_target(self, target):
       self._excludes.add_for_target(target, target.excludes)
 
   def _wrap_path_elements(self, classpath_elements):
-    return [(element[0], ClasspathEntry(*element[1:])) for element in classpath_elements]
+    wrapped_path_elements = []
+    for element in classpath_elements:
+      if len(element) != 2:
+        raise ValueError('Input must be a list of tuples containing two elements.')
```

Comment:
I think this isn't quite in the right place... The cases are:

* len == 2 and is `ClasspathEntry`
* len == 2 and is `string`
* len == 3 and are `string` and `DirectoryDigest` (we maybe want to no longer support this, and just update these callers to use `ClasspathEntry` directly)

so we either need to move this check to be nested in the `isinstance` if, or we should change line 344 to just do `ClasspathEntry(element[1]))` rather than `ClasspathEntry(*element[1:])`

Reply:
We no longer support the third case but I didn't want to remove the `*element[1:]` just in case because it does the same thing as `element[1]` when we throw an error if `element` doesn't have a length of two, but I will remove it.